Path/thalesmg

thalesmg opened the pull request:
This one is really ugly... Not only that, it seems that the non-diagonal version is not quite right... It does not match the example solution.

Also, this version ends up overwriting A and B in the resulting map. I'll have to polish that.

thalesmg committed:
Starting path - thalesmg

thalesmg committed:
First "working" version (no graphics)

thalesmg committed:
Working version with and without diagonals, and graphics

thalesmg committed:
Adding a few comments

neongreen commented:
It also fails with `*** Exception: Map.!: given key is not an element in the map` when there's no path.

neongreen commented:
Or even sometimes when there is a path:

```
A.........#.....#....
...####...#.....#....
...#........#####....
...#..#.....#.......#
...####.##......#....
#.......#............
...#........#.....#..
......#....#.#.......
...#.......#B#.#.....
.....#..####.#.......
#............#.......
```

Also it should find positions of start and end automatically instead of having them hardcoded.

thalesmg committed:
Incorporating detection of start and end nodes.

thalesmg committed:
Fixed algorithm and requirements for task.

Fixed Dijkstra's algorithm (it was visiting nodes in the wrong order);
Now the start and end points are read automatically and are shown in the result;
Now the exception is not thrown if there is no path.
